lib/graphql/card.ts

```diff
@@ -0,0 +1,101 @@
+import { Card as CardModel, Query } from "./schema";
+import { Model } from "./model";
+import { ResultPage } from "./resultPage";
+import { KontistSDKError } from "../errors";
+import { CardOptions } from "./types"
+
+
+const CARD_FIELDS = `
+  id
+  status
+  type
+  canceledAt
+  expirationDate
+  orderedAt
+  holder
+  formattedExpirationDate
+  maskedPan
+  settings {
+    contactlessEnabled
+    cardNotPresentLimits {
+      daily {
+        maxAmountCents
+        maxTransactions
+      }
+      monthly {
+        maxAmountCents
+        maxTransactions
+      }
+    }
+    cardPresentLimits {
+      daily {
+        maxAmountCents
+        maxTransactions
+      }
+      monthly {
+        maxAmountCents
+        maxTransactions
+      }
+    }
+  }
+`;
+
+const GET_CARDS = `query {
+  viewer {
+    mainAccount {
+      cards {
+        ${CARD_FIELDS}
+      }
+    }
+  }
+}`;
+
+const GET_CARD = `
+  query getCard (
+    $cardId: String,
+    $type: CardType
+  ) {
+    viewer {
+      mainAccount {
+        card(
+          filter: {
+            cardId: $cardId,
+            type: $type
+          }
+        ) {
+          ${CARD_FIELDS}
+        }
+      }
+    }
+  }
+`;
+
+export class Card extends Model<CardModel> {
+  async fetch(): Promise<ResultPage<CardModel>> {
```

Comment:
I wanted to use `fetch` to get all cards of user, but unfortunately it needs to implement `ResultPage<ModelType, FetchOptionsType>` as per our base model.

So I went with a different method, `getAll()` and `get({})`, how does it look? As user of this interface @p-janik let me know if you would like anything different or have better ideas

Reply:
Did not get it - why is it not possible to implement that interface?

Reply:
cards resolver is not paginated, GQL returns all cards, so there is no notion of cursors, edge, etc.

Reply:
hmm now taht I think about it we could still use it, just need to provide 
```
      hasNextPage: false,
      hasPreviousPage: false
```

every time without cursors 🤔 let me update

Reply:
any idea about the method name to get card by ID? just `get({ ... })` ?

Reply:
Here: ab2bcb0ca0155a5d2bfa1b46cf0163024702a7e5

Thank you for your comment 🙏